tests/typed_agent.py

```diff
@@ -281,6 +285,11 @@ def my_method(self) -> bool:
 # since deps are not set, they default to `None`, so can't be `int`
 Agent('test', tools=[Tool(foobar_plain)], deps_type=int)  # pyright: ignore[reportArgumentType,reportCallIssue]
 
+# TextOutput with RunContext
+Agent('test', output_type=TextOutput(str_to_regex_with_ctx), deps_type=int)
```

Comment:
To verify that `contravariant=True` is the right option, we should also have an example where the `deps_type` is a subclass of the type on the function's ctx, e.g. `bool`. That should work, because a function that takes an int is also able to handle a bool (which is essentially 1 or 0.